CRM/Civicase/Page/ActivityFiles.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+class CRM_Civicase_Page_ActivityFiles {
+
+  /**
+   * Download all activity files contained in a single zip file.
+   */
+  public static function downloadAll() {
+    $activity = self::getActivityFromRequest();
+
+    $zipName = self::getZipName($activity);
+    $zipDestination = self::getDestinationPath();
+    $zipFullPath = $zipDestination . '/' . $zipName;
+    $files = self::getActivityFilePaths($activity['id']);
+
+    self::createOrUpdateZipFile($zipFullPath, $files);
+    self::downloadZipFile($zipFullPath);
+  }
+
+  /**
+   * Returns the activity specified by the request. In case the request gives
+   * an invalid activity id it throws a 404 status code.
+   */
+  private static function getActivityFromRequest() {
+    $activityId = CRM_Utils_Array::value('activity_id', $_GET);
+    self::validateActivityId($activityId);
+
+    $activityResult = civicrm_api3('Activity', 'get', ['id' => $activityId]);
+
+    if ($activityResult['count'] === 0) {
+      return self::throw404StatusCode();
+    }
+
+    return CRM_Utils_Array::first($activityResult['values']);
+  }
+
+  /**
+   * Validates that the activity id was provided. If not, it returns a 404 status code.
+   *
+   * @param string|null $activityId
+   */
+  private static function validateActivityId($activityId) {
+    if (empty($activityId)) {
```

Comment:
No, I validate because `'id' => ''` would return all records, not to skip the api call. I know that if it's defined it's going to be a number for sure because passing anything else throws the following error

<img width="884" alt="screen shot 2018-09-04 at 11 10 08 am" src="https://user-images.githubusercontent.com/1642119/45088277-82474980-b0d6-11e8-97f2-60b5bb02978e.png">
`One of parameters (value "something something") was not of the type Integer`